packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -377,19 +377,26 @@ export const useGridSelection = (
   }, [apiRef, isRowSelectable, isStateControlled]);
 
   const updateColumnsPreProcessing = React.useCallback(() => {
-    const addCheckboxColumn = (columns: GridColumns) => {
-      if (!props.checkboxSelection) {
-        return columns;
-      }
-
-      const groupingColumn: GridColDef = {
+    const addCheckboxColumn = (columnsState: GridColumnsRawState) => {
```

Comment:
It is a bit confusing to call "add..." a function that can remove the checkbox column "update" or "manage" could be more appropriate

Reply:
Indeed, fixed :+1: